src/source/Signaling/StateMachine.c

```diff
@@ -778,9 +782,6 @@ STATUS fromDisconnectedSignalingState(UINT64 customData, PUINT64 pState)
 
     CHK(pSignalingClient != NULL && pState != NULL, STATUS_NULL_ARG);
 
-    // See if we need to retry first of all
-    CHK(pSignalingClient->pChannelInfo->reconnect, STATUS_SUCCESS);
```

Comment:
Purpose of this change?

Reply:
It's a bug that leads to unnecessary state transitions. If pChannelInfo->reconnect is true, we'll jump to Ready even if we have an Auth error, which thenj will make the next ConnectSync have an auth error to then force it to jump to GET_TOKEN. The skip is unnecessary.

Reply:
Ok. In that case, shouldnt the jump be conditional? I think it is ok for now, but we will have to look into making this smart.